operator/v1/types_ingress.go

```diff
@@ -1032,6 +1032,53 @@ type IngressControllerHTTPHeaderBuffer struct {
 	HeaderBufferMaxRewriteBytes int32 `json:"headerBufferMaxRewriteBytes,omitempty"`
 }
 
+// HSTSDirectiveType is an optional directive that directs HSTS Policy configuration.
+//+kubebuilder:validation:Enum=preload;includeSubDomains
+type HSTSDirectiveType string
+
+const (
+	// PreloadHSTSDirective directs the client to include hosts in its host preload list so that
+	// it never needs to do an initial load to get the HSTS header (note that this is not defined
+	// in RFC 6797 and is therefore client implementation-dependent).
+	PreloadHSTSDirective HSTSDirectiveType = "preload"
+
+	// IncludeSubDomainsHSTSDirective means the HSTS Policy applies to any subdomains of the host's
+	// domain name, when the route admission policy allows wildcards.  If wildcard routes aren't allowed,
+	// this directive has no effect.
+	IncludeSubDomainsHSTSDirective HSTSDirectiveType = "includeSubDomains"
+
+	AllHSTSHeaderPolicyScope     = "All"
+	LimitedHSTSHeaderPolicyScope = "Limited"
+)
+
+type IngressControllerHSTSHeaderPolicy struct {
+	// maxAgeSeconds is the delta time range in seconds during which hosts are regarded as HSTS hosts.
+	// If set to 0, it negates the effect and hosts are no longer regarded as HSTS hosts.
+	// maxAgeSeconds is a time-to-live value, and if this policy is not refreshed on a client, the HSTS
+	// policy will eventually expire on that client.
+	// +required
```

Comment:
Should there be `// +kubebuilder` comments here?

Reply:
Sure - the value cannot be smaller than 0 and as an int32, no larger than that.

Reply:
So then perhaps the following would make sense?

```go
// +kubebuilder:validation:Required
// +kubebuilder:validation:Minimum=0
```